src/pointer.rs

```diff
@@ -288,6 +248,36 @@ impl Pointer {
             )
             .into()
     }
+    /// Splits the `Pointer` at the given index if the character at the index is
+    /// a seperator backslash (`'/'`), returning `Some((head, tail))`. Otherwise,
+    /// returns `None`.
+    ///
+    /// For the following JSON Pointer, the following splits are possible (0, 4, 8):
+    /// ```text
+    /// /foo/bar/baz
+    /// ↑   ↑   ↑
+    /// 0   4   8
+    /// ```
+    /// All other indices will return `None`.
+    ///
+    /// ## Example
+    ///
+    /// ```rust
+    /// # use jsonptr::Pointer;
+    /// let ptr = Pointer::from_static("/foo/bar/baz");
+    /// let (head, tail) = ptr.split_at(4).unwrap();
+    /// assert_eq!(head, Pointer::from_static("/foo"));
+    /// assert_eq!(tail, Pointer::from_static("/bar/baz"));
+    ///
+    /// assert_eq!(ptr.split_at(3), None);
+    /// ```
+    pub fn split_at(&self, idx: usize) -> Option<(&Self, &Self)> {
+        if self.0.get(idx..idx + 1) != Some("/") {
+            return None;
+        }
+        let (head, tail) = self.0.split_at(idx);
```

Comment:
Yeah I think I can do the same for `split_front` and `split_back` not allocate. Originally they did because `Token` was owned, but now that it isn't I think it should be possible to mimic std.

Reply:
I think we are good - there are so few allocations on this. Your changes have been such a huge improvement on that already!

Fixing this crate was going to be next on my list after my json schema crate. I'm so grateful I don't have to do that now, especially since this turned out way better than what I would have produced!